L1Trigger/TrackFindingTracklet/src/TrackDer.cc

```diff
@@ -0,0 +1,97 @@
+
+#include "L1Trigger/TrackFindingTracklet/interface/TrackDer.h"
+
+using namespace std;
+using namespace Trklet;
+
+TrackDer::TrackDer() {
+  for (unsigned int i = 0; i < 6; i++) {
+    irinvdphi_[i] = 9999999;
+    irinvdzordr_[i] = 9999999;
+    iphi0dphi_[i] = 9999999;
+    iphi0dzordr_[i] = 9999999;
+    itdphi_[i] = 9999999;
+    itdzordr_[i] = 9999999;
+    iz0dphi_[i] = 9999999;
+    iz0dzordr_[i] = 9999999;
+
+    rinvdphi_[i] = 0.0;
+    rinvdzordr_[i] = 0.0;
+    phi0dphi_[i] = 0.0;
+    phi0dzordr_[i] = 0.0;
+    tdphi_[i] = 0.0;
+    tdzordr_[i] = 0.0;
+    z0dphi_[i] = 0.0;
+    z0dzordr_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < 3; i++) {
+    for (unsigned int j = 0; j < 3; j++) {
+      tdzcorr_[i][j] = 0.0;
+      z0dzcorr_[i][j] = 0.0;
+    }
+  }
+}
+
+void TrackDer::setIndex(int layermask, int diskmask, int alphamask, int irinv) {
+  layermask_ = layermask;
+  diskmask_ = diskmask;
+  alphamask_ = alphamask;
+  irinv_ = irinv;
+}
+
+void TrackDer::fill(int t, double MinvDt[4][12], int iMinvDt[4][12]) const {
+  unsigned int nlayer = 0;
+  if (layermask_ & 1)
+    nlayer++;
+  if (layermask_ & 2)
+    nlayer++;
+  if (layermask_ & 4)
+    nlayer++;
+  if (layermask_ & 8)
+    nlayer++;
+  if (layermask_ & 16)
+    nlayer++;
+  if (layermask_ & 32)
+    nlayer++;
+  int sign = 1;
+  if (t < 0)
+    sign = -1;
+  for (unsigned int i = 0; i < 6; i++) {
+    if (i < nlayer) {
+      MinvDt[0][2 * i] = rinvdphi_[i];
+      MinvDt[1][2 * i] = phi0dphi_[i];
+      MinvDt[2][2 * i] = sign * tdphi_[i];
+      MinvDt[3][2 * i] = sign * z0dphi_[i];
+      MinvDt[0][2 * i + 1] = sign * rinvdzordr_[i];
+      MinvDt[1][2 * i + 1] = sign * phi0dzordr_[i];
+      MinvDt[2][2 * i + 1] = tdzordr_[i];
+      MinvDt[3][2 * i + 1] = z0dzordr_[i];
+      iMinvDt[0][2 * i] = irinvdphi_[i];
+      iMinvDt[1][2 * i] = iphi0dphi_[i];
+      iMinvDt[2][2 * i] = sign * itdphi_[i];
+      iMinvDt[3][2 * i] = sign * iz0dphi_[i];
+      iMinvDt[0][2 * i + 1] = sign * irinvdzordr_[i];
+      iMinvDt[1][2 * i + 1] = sign * iphi0dzordr_[i];
+      iMinvDt[2][2 * i + 1] = itdzordr_[i];
+      iMinvDt[3][2 * i + 1] = iz0dzordr_[i];
+    } else {
+      MinvDt[0][2 * i] = rinvdphi_[i];
```

Comment:
a lot of these terms are the same as the previous case - try to avoid repetition

Reply:
simplified this